src/rules/noUnusedVariableRule.ts

```diff
@@ -33,18 +38,57 @@ class NoUnusedVariablesWalker extends Lint.RuleWalker {
     private skipParameterDeclaration: boolean;
     private skipVariableDeclaration: boolean;
 
+    // after a this.walk is completed, this.reactImport will be set to a NamespaceImport
+    // iff:
+    //
+    // + a react option has been provided to the rule and
+    // + an import of a module that matches one of OPTION_REACT_MODULES is found, to a
+    //   NamespaceImport named OPTION_REACT_NAMESPACE_IMPORT_NAME
+    //
+    // e.g.
+    //
+    // import * as React from "react/addons";
+    //
+    // If reactImport is defined when a walk is completed, we need to have:
+    //
+    // a) seen another usage of React and/or
+    // b) seen a JSX identifier
+    //
+    // otherwise a usage error will be reported
+    private reactImport: ts.NamespaceImport;
+    private reactUsed: boolean;
+    private seenJsx: boolean;
+
     constructor(sourceFile: ts.SourceFile, options: Lint.IOptions, languageService: ts.LanguageService) {
         super(sourceFile, options);
         this.languageService = languageService;
         this.skipVariableDeclaration = false;
         this.skipParameterDeclaration = false;
+        this.reactImport = undefined;
+        this.reactUsed = false;
+        this.seenJsx = false;
+    }
+
+    public walk(node: ts.Node): void {
```

Comment:
Override `visitSourceFile` instead of `walk` for actions that should happen at the end of traversal. The `walk` method is technically more generic any could be called on any node (I guess). At the least, using `visitSourceFile` is more consistent with the other rules